Microsoft.Azure.CosmosRepository/src/Providers/DefaultCosmosContainerProvider.cs

```diff
@@ -1,96 +1,101 @@
             ICosmosPartitionKeyPathProvider cosmosPartitionKeyPathProvider,
             IOptions<RepositoryOptions> options,
             ILogger<DefaultCosmosContainerProvider<TItem>> logger)
         {
-            _cosmosClientProvider = cosmosClientProvider
+            this.cosmosClientProvider = cosmosClientProvider
                 ?? throw new ArgumentNullException(
-                    nameof(cosmosClientProvider), "Cosmos client provider is required.");
+                    nameof(cosmosClientProvider), Properties.Resources.CosmosClientProviderIsRequired);
 
-            _cosmosPartitionKeyPathProvider = cosmosPartitionKeyPathProvider
+            this.cosmosPartitionKeyPathProvider = cosmosPartitionKeyPathProvider
                 ?? throw new ArgumentNullException(
-                    nameof(cosmosPartitionKeyPathProvider), "Cosmos partition key name provider is required.");
+                    nameof(cosmosPartitionKeyPathProvider), Properties.Resources.CosmosPartitionKeyNameProviderIsRequired);
 
-            _options = options?.Value
+            this.options = options?.Value
                 ?? throw new ArgumentNullException(
-                    nameof(options), "Repository options are required.");
+                    nameof(options), Properties.Resources.RepositoryOptionsAreRequired);
 
-            if (_options.CosmosConnectionString is null)
+            if (this.options.CosmosConnectionString is null)
             {
-                throw new ArgumentNullException($"The {nameof(_options.CosmosConnectionString)} is required.");
+                throw new NullReferenceException(
+                    string.Format(
+                        Properties.Resources.TheInsertNameHereIsRequired,
+                        nameof(DefaultCosmosContainerProvider<TItem>.options.CosmosConnectionString)));
             }
-            if (_options.ContainerPerItemType is false)
+
+            if (this.options.ContainerPerItemType is false)
             {
-                if (_options.DatabaseId is null)
+                if (this.options.DatabaseId is null)
                 {
-                    throw new ArgumentNullException(
-                        $"The {nameof(_options.DatabaseId)} is required when container per item type is false.");
+                    throw new NullReferenceException(
+                        string.Format(
+                            Properties.Resources.TheInsertNameHereIsRequiredWhenContainerPerItemTypeIsFalse,
+                            nameof(DefaultCosmosContainerProvider<TItem>.options.DatabaseId)));
                 }
-                if (_options.ContainerId is null)
+
+                if (this.options.ContainerId is null)
                 {
-                    throw new ArgumentNullException(
-                        $"The {nameof(_options.ContainerId)} is required when container per item type is false.");
+                    throw new NullReferenceException(
```

Comment:
I understand the formatting of the string, and the usage of the resource for storing it. Curious about the choice to change the `nameof`, using the `_options.ContainerId` works and is a lot less verbose. Also, you changed the type of the exception here, which is not desirable. I prefer to say that if part of the options are missing, then the argument is `null` as it is part of the parameter that we're validating against.